client/app/components/TimeAgo.jsx

```diff
@@ -0,0 +1,91 @@
+import moment from 'moment';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+
+const interactiveComponents = new Set();
+
+function updateInteractiveComponents() {
+  interactiveComponents.forEach(component => component.forceUpdate());
+  setTimeout(updateInteractiveComponents, 3000);
+}
+updateInteractiveComponents();
```

Comment:
Might this introduce a memory leak since the timeout never expires?
(One way to go about it is to store the timeout and then clear it when unmounting)

Reply:
tldr: No :-)

This is a single timer to update all `TimeAgo` instances. `TimeAgo` components add themselves to a queue when they are mounted and remove when unmounted, so no any memory leak. I use this pattern sometimes because it's much easier than managing timers (you have to store timer ID somewhere when you create a timer, you should set that variable to `null` when you destroy a timer because browser may reuse that ID, and so on). In this case, component instances does not need different timers, queue will contain at most ~100 elements (users list page with max. page size), so I decided to implement it in this way. The only thing I may need to change - interval (see @arikfr comment above) - updating every `3000ms` may be too often, but I don't think we need progressive timers for each component instance as in `angular-moment`.

Reply:
Yes. But is that a problem? When queue is empty it returns almost immediately, and the next tick is in 30 seconds.

Reply:
True, there's no perf hit.
It's just a red flag for me. From my experience, these things tend to come back with a vengeance 😓 But I'm cool with it if you're cool with it.

One last suggestion (but only if it's an easy implementation) - observe the Set and start/stop the timer according to its length?

Reply:
While I agree that this implementation is a bit tricky, but in this case I'm totally fine with it (even if timer will stay working while queue is empty).